Gems/ROS2/Code/Source/Manipulation/JointsManipulationComponent.cpp

```diff
@@ -232,6 +256,92 @@ namespace ROS2
         return positions;
     }
 
+    JointsManipulationRequests::JointsVelocitiesMap JointsManipulationComponent::GetAllJointsVelocities()
+    {
+        JointsManipulationRequests::JointsVelocitiesMap velocities;
+        for (const auto& [jointName, jointInfo] : m_manipulationJoints)
+        {
+            velocities[jointName] = GetJointVelocity(jointName).GetValue();
+        }
+        return velocities;
+    }
+
+    AZ::Outcome<JointEffort, AZStd::string> JointsManipulationComponent::GetJointEffort(const AZStd::string& jointName)
+    {
+        if (!m_manipulationJoints.contains(jointName))
+        {
+            return AZ::Failure(AZStd::string::format("Joint %s does not exist", jointName.c_str()));
+        }
+
+        auto jointInfo = m_manipulationJoints.at(jointName);
+        float effort {0};
+
+        if (jointInfo.m_isArticulation) {
+            bool is_acceleration_driven{false};
+            PhysX::ArticulationJointRequestBus::EventResult(
+                    is_acceleration_driven,
+                    jointInfo.m_entityComponentIdPair.GetEntityId(),
+                    &PhysX::ArticulationJointRequests::IsAccelerationDrive,
+                    jointInfo.m_axis);
+
+            if (!is_acceleration_driven) {
+
+                float stiffness{0};
+                float damping{0};
+                PhysX::ArticulationJointRequestBus::EventResult(
+                        stiffness,
+                        jointInfo.m_entityComponentIdPair.GetEntityId(),
+                        &PhysX::ArticulationJointRequests::GetDriveStiffness,
+                        jointInfo.m_axis);
+                PhysX::ArticulationJointRequestBus::EventResult(
+                        damping,
+                        jointInfo.m_entityComponentIdPair.GetEntityId(),
+                        &PhysX::ArticulationJointRequests::GetDriveDamping,
+                        jointInfo.m_axis);
+
+                float target_pos{0};
+                float position{0};
```

Comment:
Code is not formated. Please always use clang-format before submitting PR.

Reply:
Done